### PR TITLE
Update hooks to typescript

### PR DESCRIPTION
**Pull request by alvaro-ricotta**
Moving Hooks to typescript

image

### CODE REVIEW

**Review by sk-enya — APPROVED** [review]
Please test cases along with migration!

make sure we should have good coverage.

**Review by sk-enya — CHANGES_REQUESTED** [review]
Test missing please add tests!

**Comment by alvaro-ricotta** [comment]
> Please test cases along with migration!
> 
> make sure we should have good coverage.

I´m not adding test on ts migration , after migrate to tS i´ll implemente test one by one and push it. it´s just for make ts migration thats all.

File: src/hooks/useDisconnect/index.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+          setNetwork({
+            networkType,
+            network,
+            networkIcon,
+            name,
+          })
+        )
+        dispatch(setActiveNetwork())
+      }
+    }
```
Comment (sk-enya) [inline src/hooks/useDisconnect/index.ts:45]: Looks like this lines has not been covered in tests can you please take look once.

Screenshot 2023-11-07 at 2 51 14 AM

**Review by sk-enya — CHANGES_REQUESTED** [review]

**Comment by sk-enya** [comment]
Looks like branch is not update and missing test cases for  `useGoogleAnalytics` if you are already working that's great!

File: src/hooks/useGoogleAnalytics/index.ts
Comment (sk-enya) [inline src/hooks/useGoogleAnalytics/index.ts:1]: Are you working writing test cases for this hooks as well?

File: src/hooks/useInterval.ts
Comment (sk-enya) [inline src/hooks/useInterval.ts:1]: Are you working writing test cases for this hooks as well?

**Review by sk-enya — APPROVED** [review]
Change looks good!
try running test with below command so you will get to know how much code coverage we have.
```
yarn test:w src/hooks/useDisconnect/index.test.tsx --coverage --collectCoverageFrom=src/hooks/useDisconnect/index.ts
```

**Pull request merged**